src/pages/loading-waiting-states/progress-bar/index.tsx

```diff
@@ -0,0 +1,234 @@
+    <LinearProgress variant="determinate" {...props} />
+);
+const uploadFileList: FolderItem[] = [];
+
+const useStyles = makeStyles((theme: Theme) => ({
+    container: {
+        padding: `${theme.spacing(2)}px ${theme.spacing(2)}px`,
+        maxWidth: 600,
+        margin: '0 auto',
+    },
+    uploadButtonContainer: {
+        textAlign: 'right',
+        paddingBottom: theme.spacing(2),
+    },
+    closeButtonContainer: {
+        color: Colors.black[50],
+        borderColor: Colors.black[50],
+    },
+    cardContentStyle: {
+        padding: 0,
+    },
+    formControlStyle: {
+        width: '100%',
+    },
+    radioLabel: {
+        display: 'flex',
+    },
+    iconStyle: {
+        fill: Colors.gray[500],
+        margin: `0 ${theme.spacing(2)}px 0 ${theme.spacing(1)}px`,
+    },
+    formLabel: {
+        margin: 0,
+        width: '100%',
+        padding: theme.spacing(1),
+        borderBottom: `1px solid ${Colors.gray[50]}`,
+        '&:last-child': {
+            borderBottom: 'none',
+        },
+    },
+    toolbarGutters: {
+        padding: `0 ${theme.spacing(2)}px`,
+    },
+    placementOfList: {
+        position: 'absolute',
+        bottom: theme.spacing(3),
+        right: theme.spacing(3),
+    },
+    infoListItemStyle: {
+        marginBottom: theme.spacing(2),
+        '&:last-child': {
+            marginBottom: 0,
+        },
+    },
+    subTitle: {
+        color: Colors.black[200],
+    },
+}));
+const createFileItem = (): FolderItem => ({
+    id: `${Math.random() * 100}`,
+    name: 'PX Blue is Awesome.pdf',
+    progress: 0,
+    status: `Uploading (0%)`,
+});
+
+export const ProgressBar = (): JSX.Element => {
+    const dispatch = useDispatch();
+    const classes = useStyles();
+    const [list, setUploadFileList] = useState<FolderItem[]>(uploadFileList);
+
+    const uploadFile = useCallback((): void => {
+        setUploadFileList((oldList) => [...oldList, createFileItem()]);
+    }, [list, setUploadFileList]);
+
+    useEffect(() => {
+        const progressInterval = setInterval(() => {
+            const newList = [...list];
+            for (let i = 0; i < list.length; i++) {
+                if (list[i].progress < 100) {
+                    const newItem: FolderItem = {
+                        ...list[i],
+                        progress: list[i].progress + 1,
+                        status: `Uploading (${list[i].progress + 1})`,
+                    };
+                    newList[i] = newItem;
+                } else {
+                    const newItem: FolderItem = {
+                        ...list[i],
+                        status: `Complete`,
+                    };
+                    newList[i] = newItem;
+                    setTimeout(() => {
+                        setUploadFileList((oldList) => oldList.filter((item) => item.id !== list[i].id));
+                    }, 10000);
```

Comment:
10 seconds feels like a lifetime...is that what we used on Angular?

Reply:
We can even make it a random time between 5s to 10s to suggest different file sizes :P

Reply:
This one is the time to dismiss the snackbar, which should probably be the same for all of them. But I agree to having some more randomization with the upload speed...for that we could just change the percentage update each cycle from 1 to some random value between like 1-5 — then each file will finish uploading in between 4 and 20 seconds.

Reply:
My bad. I didn't read it carefully enough. For autohide/dismissing, 3 - 5 might be ideal. We can have a follow-up story to match ng and react. Ekta, can you make one?

I like your idea about file upload speed.